lib/ansiblelint/formatters/__init__.py

```diff
@@ -1,20 +1,46 @@
 """Output formatters."""
+import os
+from pathlib import Path
+
 try:
     from ansible import color
 except ImportError:
     from ansible.utils import color
-from ansiblelint.utils import normpath
 
 
-class Formatter(object):
+class BaseFormatter:
+    """Formatter of ansible-lint output.
+
+    Base class for output formatters.
+
+    Args:
+        base_dir (str|Path): reference directory against which display relative path.
+        display_relative_path (bool): whether to show path as relatvie or absolute
+    """
+
+    def __init__(self, base_dir, display_relative_path):
+        if isinstance(base_dir, Path):
+            base_dir = str(base_dir.absolute())
+
+        self._base_dir = base_dir if display_relative_path else None
+
+    def _format_path(self, path):
+        if isinstance(path, Path):
+            path = str(path)
```

Comment:
I don't think we need to do the conversion. It should be done when producing the output and will probably happen automatically anyway.

Reply:
I think we should do the opposite and use pathlib representation wherever possible instead.